functions/Restore-DbaBackup.ps1

```diff
@@ -0,0 +1,215 @@
+function Restore-DbaBackup
+{
+<#
+.SYNOPSIS 
+Restores a SQL Server Database from a set of backupfiles
+
+.DESCRIPTION
+Scans a given folder for Full, Differential and Log backups. 
+OR
+Takes a set of folder arrays and processes them for Full, Differential and Log backups.
+
+These are then filtered and restored to a specified SQL Server intance and file location
+
+The backup LSN chain and RecoveryForkID will also be checked to ensure the restore is valid
+
+It can also generate restore scripts, both as part of a restore or as it's only action
+
+.PARAMETER SqlServer
+The SQL Server instance. 
+
+.PARAMETER SqlCredential
+Allows you to login to servers using SQL Logins as opposed to Windows Auth/Integrated/Trusted. 
+
+.PARAMETER Path
+Path to SQL Server backup file. This can be a full, differential or log backup file.
+
+.PARAMETER Files
```

Comment:
I agree with the exception, but it's not applicable here.

We can't use FileList because that's already there, as the param that says to return the list of files.
And even if it wasn't, I definitely don't want "FileList" as the imput param. That's not better than "Files".
"File" follows the same reasoning as "Path", and hundreds of other parameters.